Allow rlang-style lambda expressions in stat_summary functions

dkahle opened the pull request.

thomasp85 commented on R/stat-summary-bin.R:
```diff
@@ -96,7 +96,11 @@ make_summary_fun <- function(fun.data, fun, fun.max, fun.min, fun.args) {
+      fun.data <- rlang::as_function(fun.data)
+    } else {
+      fun.data <- match.fun(fun.data)
+    }
```
I think you can just use `as_function()` for everything as it behaves correctly in all cases... We also import rlang so no need to prefix it

yutannihilation commented:
> In general I think it is safe to just substitute `match.fun()` with `as_function()`

Agreed.

thomasp85 commented:
@dkahle do you want to finish this off?

dkahle commented:
@thomasp85 Yep! Is that what you're looking for?

thomasp85 commented on R/stat-summary-2d.r:
```diff
@@ -100,7 +100,7 @@ StatSummary2d <- ggproto("StatSummary2d", Stat,
     ybin <- cut(data$y, ybreaks, include.lowest = TRUE, labels = FALSE)
 
-    if (is.formula(fun)) fun <- rlang::as_function(fun)
+    if (is.formula(fun)) fun <- as_function(fun)
```
I think you can just remove the `is.formula` check and call `as_function()` unconditionally... same with the other two places

thomasp85 commented:
Apart from the small comment you just need to fix the conflict in the news file, and it is good to go... thanks

dkahle commented:
Hm. I pulled the news from upstream/master, updated it, and it still lists it as a conflict. What's the best way to resolve that? (Other than not wait two months before addressing your comments, that is.:)

thomasp85 commented:
You should merge the master branch into this branch (make sure master is current)

And don’t worry about time-to-respond. We all have shifting focus🙂

thomasp85 commented:
You should fix the merge conflict after merging in master. You still have the conflict in the NEWS file (the weird `<<<<< HEAD` notation)

Are you familiar with fixing merge conflicts or should I walk you through it?

dkahle commented:
Just seeingx your second comment. I am familiar with merge conflicts; I thought I removed all merge conflict notes - did I miss something?

thomasp85 commented on NEWS.md:
```diff
@@ -1,5 +1,19 @@
 # ggplot2 (development version)
 
+<<<<<<< HEAD
```
There is something here🙂

dkahle replied:
Why yes, yes there is. 😂 I feel like this is the second time that my commit message should have a facepalm emoji.  

I actually searched the project for <<<<, but I realize now I was only searching through the R script files... Stay tuned. Thanks!

thomasp85 commented:
Thank you!
